dymos/examples/cannonball/test/test_phase_linkage_compliance.py

```diff
@@ -501,7 +512,105 @@ def test_link_bounded_times_final_to_initial(self):
+        coast.add_state('theta', fix_initial=False, fix_final=False, defect_scaler=100.0,
+                        rate_source='theta_dot', units='rad')
+        coast.add_state('vr', fix_initial=False, fix_final=False, defect_scaler=100.0,
+                        rate_source='vr_dot', units='DU/TU')
+        coast.add_state('vt', fix_initial=False, fix_final=False, defect_scaler=100.0,
+                        rate_source='vt_dot', units='DU/TU')
+        coast.add_state('accel', fix_initial=True, fix_final=True, ref=1.0E-12, defect_ref=1.0E-12,
+                        rate_source='at_dot', units='DU/TU**2')
+        coast.add_state('deltav', fix_initial=False, fix_final=False,
+                        rate_source='deltav_dot', units='DU/TU')
+        coast.add_control('u1', opt=False, val=0.0, units='deg')
+        coast.add_parameter('c', opt=False, val=1.5, units='DU/TU')
+
+        # Third Phase (burn)
+
+        burn2 = dm.Phase(ode_class=FiniteBurnODE, transcription=dm.GaussLobatto(num_segments=3, order=3))
+
+        traj.add_phase('burn2', burn2)
+
+        burn2.set_time_options(initial_bounds=(0.5, 20), duration_bounds=(.5, 10), initial_ref=10, units='TU')
+        burn2.add_state('r', fix_initial=False, fix_final=True,
+                        rate_source='r_dot', units='DU')
+        burn2.add_state('theta', fix_initial=False, fix_final=False,
+                        rate_source='theta_dot', units='rad')
+        burn2.add_state('vr', fix_initial=False, fix_final=True,
+                        rate_source='vr_dot', units='DU/TU')
+        burn2.add_state('vt', fix_initial=False, fix_final=True,
+                        rate_source='vt_dot', units='DU/TU')
+        burn2.add_state('accel', fix_initial=False, fix_final=False, defect_ref=1.0E-6,
+                        rate_source='at_dot', units='DU/TU**2')
+        burn2.add_state('deltav', fix_initial=False, fix_final=False,
+                        rate_source='deltav_dot', units='DU/TU')
+        burn2.add_control('u1', rate_continuity=True, rate2_continuity=True,
+                          units='deg', scaler=0.01, lower=-30, upper=30)
+        burn2.add_parameter('c', opt=False, val=1.5, units='DU/TU')
+
+        burn2.add_objective('deltav', loc='final')
+
+        # Link Phases
+        traj.link_phases(phases=['burn1', 'coast', 'burn2'],
+                         vars=['time', 'r', 'theta', 'vr', 'vt', 'deltav', 'c'])
+
+        with self.assertRaises(ValueError) as e:
+            p.setup()
+
+        self.assertEqual(str(e.exception), 'Invalid linkage in Trajectory traj: Cannot link final value of "c" in '
+                                           'burn1 to initial value of "c" in coast. Values on both sides of the '
+                                           'linkage are fixed and the linkage is enforced via constraint. Either link '
+                                           'the variables via connection or make the variables design variables on at '
```

Comment:
I wonder if this is still a little confusing. Maybe say "Either set connected=True or set fix_initial or fix_final to False.  I guess it is a matter of telling them what to do in dymos, rather than telling them what to do conceptually.  It is up to you though, I don't know how most other people see it.

Reply:
I went a more conceptual route here because there's more than one way to do it depending on what is being linked. Fundamentally I think the user has to know if their output can be moved by the optimizer on either side of the constraint.

If it's something that has the option to be a design variable, that's a bit easier to deal with, but it can still be a difference between what kind of design variable it is.  States and controls have `fix_initial=False` and `fix_final=False`.  States, controls, and parameters have `opt=True` that in some cases obviates fix_initial and fix_final.  Time has `fix_initial` and `fix_duration`, and everything also has bounds that could be optionally implemented that could effectively fix their values anyway (lower=0, upper=0).